### PR TITLE
Add Bidirectional Recurrent Neural Network tutorial

### PR DESCRIPTION
**Pull request by mfl28**
Adds the bidirectional RNN tutorial.

### CODE REVIEW

**Comment by mfl28** [comment]
Yes I think that would certainly be a good idea! We could create something like a Learner/Trainer class that holds a model and training/validation data loaders and optimizer etc. and has train()/validate() methods were we put the currently repeated code from the main.cpp's.

```c++
// ...
Trainer trainer(model, optimizer, training_loader, validation_loader ...);
// Then just do
trainer.train(num_epochs)
// and
trainer.validate();
```
In the tutorials for the recurrent networks there are some differences in how the batch data is processed (before it is send to the appropriate device) and in another tutorial learning rate decay is used. We could register these things as callbacks in the Trainer class.

**Comment by prabhuomkar** [comment]
@mfl28 looks idea is good, but I will need some time to think on this. Will plan this weekend.
I will merge this for now. Let's do the refactoring with a new CL and its PR.
